packages/eas-build-job/src/common.ts

```diff
@@ -35,10 +35,10 @@ export enum BuildTrigger {
 
 export type ArchiveSource =
   | { type: ArchiveSourceType.NONE }
-  | { type: ArchiveSourceType.S3; bucketKey: string }
-  | { type: ArchiveSourceType.GCS; bucketKey: string }
-  | { type: ArchiveSourceType.URL; url: string }
-  | { type: ArchiveSourceType.PATH; path: string }
+  | { type: ArchiveSourceType.S3; bucketKey: string; metadataLocation?: string }
+  | { type: ArchiveSourceType.GCS; bucketKey: string; metadataLocation?: string }
+  | { type: ArchiveSourceType.URL; url: string; metadataLocation?: string }
```

Comment:
What does the `metadataLocation` mean in the context of the `URL` archive?

It makes sense for buckets because this is the path to the metadata file in the bucket, but I'm not sure about the `URL` and `PATH` archive sources.

One more Q because I didn't work with this code for a while and my knowledge got rusty: Does `bucketKey` === path to project tarball?

Reply:
+1 for making keys archive-source-type-specific, just in case (without it I think it'd be too easy to write sth like `await fetch(archiveSource.metadataLocation)`, which sometimes would work and sometimes would fail).

Reply:
yeah, I thought that eventually we may implement something for URL archive source, but I'll just remove it. YAGNI